### ISSUE TITLE
[Feature] Add environment key to the payload patch

### ISSUE THREAD

**Issue by N3TC4T** [XRPL-Labs/XUMM-Issue-Tracker#402]
Right now when a payload is submitted with XUMM we include the node type and url and also the result in the patch payload as below:

```
dispatched: {
  to: submitResult.node,
  nodetype: submitResult.nodeType,
  result: submitResult.engineResult,
}
```

As we are populating the Account Sequence, Fee etc... base on connected node it make sense to always include node type and url in the patch payload as below.
Note: the dispatched key will be omitted if transaction is not submitted with XUMM.

```
{
 environment: {
   nodeuri: '',
   nodetype: ''
 },
 dispatched: {
  to: submitResult.node,
  nodetype: submitResult.nodeType,
  result: submitResult.engineResult
 }
}
```

**Comment by WietseWind** [comment]
@N3TC4T Do you mean (second sample):

```
{
 environment: {
   nodeuri: submitResult.node,
   nodetype: submitResult.nodeType
 },
 dispatched: {
  to: submitResult.node,
  nodetype: submitResult.nodeType,
  result: submitResult.engineResult
 }
```

**Comment by WietseWind** [comment]
Done, deployed:
<img width="410" alt="image" src="https://user-images.githubusercontent.com/4756161/174311653-bcb6c747-1469-4265-ba27-afe8a41d89c5.png">

**Comment by N3TC4T** [comment]
Included in `2.3.1`
